pkg/build/controller/factory/factory.go

```diff
@@ -261,6 +260,45 @@ func (factory *ImageChangeControllerFactory) Create() controller.RunnableControl
 	}
 }
 
+type BuildConfigControllerFactory struct {
+	Client                  osclient.Interface
+	BuildConfigInstantiator buildclient.BuildConfigInstantiator
+	// Stop may be set to allow controllers created by this factory to be terminated.
+	Stop <-chan struct{}
+}
+
+// Create creates a new ConfigChangeController which is used to trigger builds on creation
+func (factory *BuildConfigControllerFactory) Create() controller.RunnableController {
+	queue := cache.NewFIFO(cache.MetaNamespaceKeyFunc)
+	cache.NewReflector(&buildConfigLW{client: factory.Client}, &buildapi.BuildConfig{}, queue, 2*time.Minute).RunUntil(factory.Stop)
+
+	bcController := &buildcontroller.BuildConfigController{
+		BuildConfigInstantiator: factory.BuildConfigInstantiator,
+	}
+
+	return &controller.RetryController{
+		Queue: queue,
+		RetryManager: controller.NewQueueRetryManager(
+			queue,
+			cache.MetaNamespaceKeyFunc,
+			func(obj interface{}, err error, retries controller.Retry) bool {
+				bc := obj.(*buildapi.BuildConfig)
+				if maxRetries > retries.Count {
```

Comment:
@soltysh it does look that way...and all our controllers are doing that.  i'm a bit puzzled since i know i've seen it retry and i think we even may have some retry tests.  Looks like we have it right in some places and backwards in others :(

@csrwng can you fix this here and in the other spots in factory.go?